File: snac.ts
export const SNACPattern: string = "^\\d+$"; // SNAC IDs are numeric
export const reSNAC: RegExp = new RegExp(SNACPattern);

/**
 * normalizeSNAC
 * @param id, a string holding the SNAC identifier
 * @returns normalized SNAC identifier as a string.
 */
export function normalizeSNAC(id: string): string {
  return id.trim();
}

/**
 * validateSNAC
 * @param id, a string holding the SNAC identifier
 * @returns true if valid, false otherwise
 */
export function validateSNAC(id: string): boolean {
  const normalizedId = normalizeSNAC(id);
  return reSNAC.test(normalizedId);
}

File: snac.go
package metadatatools

import (
	"regexp"
	"strings"
)

// SNACPattern matches a normalized SNAC identifier, which is numeric.
const SNACPattern = `^\d+$`

// reSNAC is the compiled form of SNACPattern.
var reSNAC = regexp.MustCompile(SNACPattern)

// NormalizeSNAC trims whitespace from id.
func NormalizeSNAC(id string) string {
	return strings.TrimSpace(id)
}

// ValidateSNAC normalizes id then validates it against SNACPattern.
func ValidateSNAC(id string) bool {
	return reSNAC.MatchString(NormalizeSNAC(id))
}
